### ISSUE TITLE
[BUG] Validator Message Group iteration incorrect behavior after unsetting a message index

### ISSUE THREAD

**Issue by remitoffoli** [phalcon/cphalcon#12455]
Hi, I just ran into a bug trying to iterate a Phalcon\Validation\Message\Group after having unset a message index. Here is an exemple :

```php
use Phalcon\Validation;
use Phalcon\Validation\Message;

$validation = new Validation();

// adds messages
$validation->appendMessage(new Message('error a', 'myField', 'MyValidator'));
// [0] => 'error a'
$validation->appendMessage(new Message('error b', 'myField', 'MyValidator'));
// [0] => 'error a',  [1] => 'error b'

// removes the last message (for whatever reason, I've mine...)
$messages = $validator->getMessages();
unset($messages[count($messages)-1]); // you can't array_splice because Group isn't an array... :(
// [0] => 'error a'

// adds a new message
$validation->appendMessage(new Message('error c', 'myField', 'MyValidator'));
// [0] => 'error a',  [2] => 'error c'

// now the last message isn't in the range available by the foreach...
foreach ($messages as $msg) {
    echo $msg ."<br>";
}
// will only echo the 'error a'
```

FYI, if you have to iterate with this problem you cant use the following hack.
```php
$messages = (array) $messages;
$messages = $messages["\0*\0_messages"];
// more informations about the \0*\0 here :
// https://coderwall.com/p/36h6qa/converting-php-objects-to-arrays-to-read-private-members
```

The problem can easily be fixed by replacing ```unset this->_messages[index];``` by an ```array_splice``` in the following function of https://github.com/phalcon/cphalcon/blob/master/phalcon/validation/message/group.zep#L115

Just like that :

```zep
public function offsetUnset(index)
{
    if isset this->_messages[index] {
        array_splice(this->_messages, index, 1);
    }
    return false;
}
```
### Details

* Phalcon version: 3.0.1
* PHP Version: 5.6.24
* Operating System: Windows 10
* Installation type: installing windows dll
* Zephir version (if any):
* Apache
* Other related info (Database, table schema): nope

**Comment by stamster** [comment]
That should not work that way. 
Use `offsetUnset()` method instead.
https://docs.phalconphp.com/en/3.0.1/api/Phalcon_Validation_Message_Group.html
https://github.com/phalcon/cphalcon/blob/master/phalcon/validation/message/group.zep#L106

**Comment by Jurigag** [comment]
But unset should work too. It's implementing ArrayAccess. When doing unset it's calling offsetUnset anyway.

**Comment by remitoffoli** [comment]
@Jurigag is right.

In fact, I thinks you should not use directly ```offsetUnset()```. PHP calls it when you use ```unset(...)``` : http://php.net/manual/en/class.arrayaccess.php

Have a look at the code exemple just below the method documentation ;)
https://docs.phalconphp.com/en/3.0.1/api/Phalcon_Validation_Message_Group.html

**Comment by gguridi** [comment]
Yes, the change is working. I've added a pull request for the change after checking that it's working properly. Not sure if there is another pull request already about this, but just in case I went ahead.

**Comment by sergeyklay** [comment]
Fixed in the `3.0.x` branch.
